src/core/workspace.c

```diff
@@ -216,12 +236,31 @@ struct workspace *workspace_list_get_focused(const struct workspace_list *list)
         return list->workspaces[list->active_index];
 }
 
+struct workspace *
+workspace_list_find_client_workspace(const struct workspace_list *list,
+                                     const struct client *client)
+{
+        struct map_entry *entry = map_get(list->client_map, &client->window);
+
+        if (entry == NULL) {
+                return NULL;
+        }
+
+        size_t index = *(size_t *)entry->value;
+
+        return list->workspaces[index];
+}
+
 void workspace_list_destroy(struct workspace_list *workspace_list)
 {
         if (workspace_list->theme != NULL) {
                 client_theme_destroy(workspace_list->theme);
         }
 
+        if (workspace_list->client_map->bucket_count > 0) {
```

Comment:
this will leak if there are no items - we need to free the memory allocated for the map, not just the map items

Reply:
![2020-04-18-122513_925x493_selection](https://user-images.githubusercontent.com/2308484/79669330-abafce80-816f-11ea-9534-0c85df30d0e8.png)